src/server/src/main/java/io/cassandrareaper/storage/MemoryStorage.java

```diff
@@ -218,8 +219,10 @@ public RepairUnit addRepairUnit(RepairUnit.Builder repairUnit) {
   }
 
   @Override
-  public Optional<RepairUnit> getRepairUnit(UUID id) {
-    return Optional.fromNullable(repairUnits.get(id));
+  public RepairUnit getRepairUnit(UUID id) {
+    RepairUnit unit = repairUnits.get(id);
+    Preconditions.checkArgument(null != unit);
```

Comment:
It throws an `IllegalArgumentException`.  Same as PostgresStorage and CassandraStorage.

Reaper can likely fail to start. Otherwise it will throw a unchecked exception, indeed possibly killing threads (although we've guarded the scheduler threads better now).

Originally I thought about using `assert` statements, and recreate on-the-fly missing repair units when assertions were not enabled (ie production).
But I realised this was just taking us down the _Eventually Consistent_ pathway, and it should be either strictly consistent and fail-fast, or eventual and healing, but not a mix between the two.